src/endpoint/s3/ops/s3_post_object_select.js

```diff
@@ -0,0 +1,261 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../../util/debug_module')(__filename);
+const S3Error = require('../s3_errors').S3Error;
+const s3_utils = require('../s3_utils');
+const http_utils = require('../../../util/http_utils');
+const { Transform } = require('readable-stream');
+const nb_native = require('../../../util/nb_native');
+const crc32 = require('crc-32');
+
+var intToNetworkByteArray = function(/*int*/long) {
+    var byteArray = [0, 0, 0, 0];
+
+    for (var index = byteArray.length - 1; index >= 0; --index) {
+        var byte = long & 0xff;
+        byteArray [ index ] = byte;
+        long = (long - byte) / 256 ;
+    }
+
+    return Buffer.from(byteArray);
+};
+
+class S3SelectStream extends Transform {
+
+    constructor(opts) {
+        super(opts);
+        this.CephS3lect = new (nb_native().CephS3lect)(opts);
```

Comment:
when we build without BUILD_S3SELECT=0 this CephS3select would be undefined and this will fail miserably. Can you add a more explicit check in the beginning of the operation that will fail but reply a more easy to understand error that this library was not built and then fail?